spec/features/tasks/edit_spec.rb

```diff
@@ -1,34 +1,39 @@
 require 'rails_helper'
+include TasksSpecSupport
 
 feature "Editing a task" do
-  let!(:task) { Task.create(name: "Test my app", completed: false) }
+  let(:user) { create(:user) }
+  let(:list) { create(:list, user: user) }
+  let!(:task) { Task.create(name: "Test my app", completed: false, list_id: list.id) }
+
+  before do
+    login_as user
+  end
 
   scenario "redirects to the tasks index page on success" do
-    visit tasks_path
-    click_on "Edit"
+    visit list_tasks_path(list.id)
+    click_on "Edit this task"
     expect(page).to have_content("Editing task")
 
-    fill_in "Name", with: "Test my app (updated)"
-    click_button "Update"
+    fill_in_form('Name', 'Test my app (updated)', 'Save')
 
     expect(page).to have_content("Tasks")
     expect(page).to have_content("Test my app (updated)")
   end
 
   scenario "displays an error when no name is provided" do
-    visit edit_task_path(task)
-    fill_in "Name", with: ""
-    click_button "Save"
+    visit edit_list_task_path(list_id: task.list_id, id: task.id)
```

Comment:
great that you tested for validations, one thing i am always thinking over in feature specs is the balance btw 1 scenario having a single responsibility vs. efficiency. 

Wondering what your thoughts are on that, and the pros/cons of testing this validation as part of the happy path scenario above.

Reply:
I agree that having a single responsibility is important. I believe that it should be the cases for most scenarios, but in real life there are cases where efficiency matters a lot. I think it could just be an overkill to write another scenario test when the test is simple and straight forward like the one above with not many lines of codes.